BookUniverseProject/BookUniverseProject/Controllers/BaseController.cs

```diff
@@ -0,0 +1,34 @@
+using BookUniverse.Application.MediatR.ResultVariations;
+using FluentResults;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+
+namespace BookUniverseProject.Controllers
+{
+    public class BaseController : Controller
+    {
+        private IMediator? _mediator;
+
+        public BaseController()
+        {
+        }
+
+        protected IMediator Mediator => _mediator ??= HttpContext.RequestServices.GetService<IMediator>()!;
+
+        protected ActionResult<T> HandleResult<T>(Result<T> result)
+        {
+            if (result.IsSuccess)
+            {
+                if (result is NullResult<T>)
+                {
+                    return Ok(result.Value);
+                }
+
+                return (result.Value is null) ?
+                    NotFound("Not Found") : Ok(result.Value);
+            }
+
+            return BadRequest(result.Reasons);
+        }
+    }
+}
```

Comment:
Take into consideration last new lines issue. Try to add them at the end of files.
Read more here:
https://gist.github.com/camh-/1bebfcff1b0f814e9b191edc60d5206b

https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log